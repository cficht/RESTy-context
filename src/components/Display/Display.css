.Display {
  margin: 20px;
  width: 80%;
  border: 5px solid black;
  border-radius: 10px;
  overflow: auto;
  background-color: #1D2021;
}
