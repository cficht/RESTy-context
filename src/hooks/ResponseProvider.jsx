import React, { createContext, useReducer, useContext } from 'react';
import PropTypes from 'prop-types';

const ResponseContext = createContext();

const initialState = {
  url: '',
  method: '',
  body: '',
  res: {},
  requests: []
};

export function reducer(state, action) {
  switch(action.type) {
    case 'SET_URL':
      return { ...state, url: action.payload };
    case 'SET_METHOD':
      return { ...state, method: action.payload };
    case 'SET_BODY':
      return { ...state, body: action.payload };
    case 'SET_RES':
      return { ...state, res: action.payload };
    case 'ADD_REQUESTS':
      return { ...state, requests: [...state.requests, action.payload] };
    default:
      return state;
  }
}

export const ResponseProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <ResponseContext.Provider value={{ state, dispatch }}>
      {children}
    </ResponseContext.Provider>
  );
};

ResponseProvider.propTypes = {
  children: PropTypes.node
};

export const useGlobalState = () => {
  const { state } = useContext(ResponseContext);
  return state;
};

export const useDispatch = () => {
  const { dispatch } = useContext(ResponseContext);
  return dispatch;
};

export const useUrl = () => {
  const { url } = useGlobalState();
  return url;
};

export const useMethod = () => {
  const { method } = useGlobalState();
  return method;
};

export const useBody = () => {
  const { body } = useGlobalState();
  return body;
};

export const useRes = () => {
  const { res } = useGlobalState();
  return res;
};

export const useRequests = () => {
  const { requests } = useGlobalState();
  return requests;
};
