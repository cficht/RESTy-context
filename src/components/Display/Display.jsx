import React from 'react';
import ReactJson from 'react-json-view';
import { useRes } from '../../hooks/ResponseProvider';
import styles from './Display.css';

const Display = () => {
  const res = useRes();
  
  return (
    <section className={styles.Display}>
      <ReactJson src={res.headers} theme="tomorrow" name="Headers"/>
      <ReactJson src={res.response} theme="tomorrow" name="Response"/>
    </section>
  );
};

export default Display;
