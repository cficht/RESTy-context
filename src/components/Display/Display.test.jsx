import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';
import { ResponseProvider } from '../../hooks/ResponseProvider';

describe('Display component', () => {
  it('renders Display', () => {
    const wrapper = shallow(
      <ResponseProvider>
        <Display />
      </ResponseProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
