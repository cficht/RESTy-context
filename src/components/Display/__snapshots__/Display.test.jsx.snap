// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "handleChange": [Function],
      "handleClear": [Function],
      "handleClick": [Function],
      "handleSubmit": [Function],
      "state": Object {
        "auth": "none",
        "body": "",
        "method": "",
        "password": "",
        "requests": Array [],
        "res": Object {},
        "token": "",
        "url": "",
        "username": "",
      },
    }
  }
>
  <Display />
</ContextProvider>
`;
